lib/ts/recipe/emailpassword/types.ts

```diff
@@ -348,9 +360,19 @@ export type SignUpThemeProps = ThemeBaseProps & {
      * Call Sign Up API.
      */
     callAPI: (fields: APIFormField[]) => Promise<SignUpThemeResponse>;
+
+    /*
+     * Verify if an email exist.
+     */
+    validateEmail: (value: string) => Promise<string | undefined>;
```

Comment:
Why are we passing this as a prop to the theme? Shouldn't there be one function that is passed that takes the id and value of a field and returns `undefined | string`?

In the feature component, this function should then have an if around the id and pass the appropriate validator on it. In case of email, if should also check that the email does not exist.